### DISCUSSION TITLE
cgo support?

### DISCUSSION THREAD

**Discussion by Julio-Guerra** [traefik/yaegi#1274] [Q&A]
#### Proposal

I guess CGO isn't supported as it is based on compile-time code generation and low-level Go ABI things, but as I couldn't clearly find the answer, I thought it would be worth asking.

#### Background

Given Traefik plugins are now based on Yaegi, advanced middleware plugins such as [web-app firewalls](https://github.com/sqreen/go-libsqreen/tree/master/waf) may be Go wrappers of C libraries.

Some of our users would like to have us in their Traefik proxy but, afaik, being limited to pure Go makes it impossible for now.

**Comment by mvertes** [comment]
As stated in the [README](https://github.com/containous/yaegi/blob/master/README.md#limitations), using CGO at interpreter level is not supported. It is always possible to wrap a pre-compiled package and export it to the interpreter by `Use` method (as it is done for the standard library), but that implies adding a new dependency to the host binary (in the plugin case, traefik).

**Reply by Julio-Guerra** [reply]
@mvertes, I'm back on this idea ^^
What about starting a side-car process having this C++ WAF running along with an API I could call from the plugin?
I wonder how could I start this side-process automatically from Traefik or the plugin (os/exec being forbidden afaik).
